dataland-frontend/tests/e2e/specs/lksg/EditButtonWithMultipleReportingPeriods.ts

```diff
@@ -0,0 +1,60 @@
+import { DataTypeEnum, LksgData, StoredCompany } from "@clients/backend";
+import { describeIf } from "@e2e/support/TestUtility";
+import { getKeycloakToken } from "@e2e/utils/Auth";
+import { generateDummyCompanyInformation, prepareUniqueCompany, uploadCompanyViaApi } from "@e2e/utils/CompanyUpload";
+import { admin_name, admin_pw } from "@e2e/utils/Cypress";
+import { prepareFixture } from "@e2e/utils/EuTaxonomyFinancialsUpload";
+import { uploadOneLksgDatasetViaApi } from "@e2e/utils/LksgUpload";
+import { FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+
+describeIf(
+  "As a user, I expect to be able to edit datasets with multiple reporting periods",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+    dataEnvironments: ["fakeFixtures"],
+  },
+  function () {
+    let testData: FixtureData<LksgData>;
+    const testCompany = prepareUniqueCompany("company-for-testing-edit-button");
+
+    before(function () {
+      const fixtureType = "CompanyInformationWithLksgPreparedFixtures";
+      prepareFixture<LksgData>(fixtureType).then((preparedFixtures) => {
+        testData = getPreparedFixture("LkSG-date-2023-04-18", preparedFixtures);
+      });
+    });
+
+    it("Check whether Edit Data button has dropdown with 2 different Reporting Periods", () => {
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompany.companyName)).then(
+          async (storedCompany) => {
+            cy.ensureLoggedIn(admin_name, admin_pw);
+
+            await uploadOneLksgDatasetViaApi(token, storedCompany.companyId, "2022", testData.t);
+            await uploadOneLksgDatasetViaApi(token, storedCompany.companyId, "2021", testData.t);
+
+            testEditDataButton(storedCompany);
+          }
+        );
+      });
+    });
+  }
+);
+
+/**
+ * Tests that the item was added and is visible on the QA list
+ * @param storedCompany details of the company that was created
+ */
+function testEditDataButton(storedCompany: StoredCompany): void {
+  cy.visitAndCheckAppMount(`/companies/${storedCompany.companyId}/frameworks/${DataTypeEnum.Lksg}`);
+
+  cy.get('[data-test="editDatasetButton"').should("exist").click();
+  cy.get('[data-test="select-reporting-period-dialog"')
+    .should("exist")
+    .get('[data-test="reporting-periods"')
+    .last()
+    .should("contain", "2021")
+    .click();
+
+  cy.get('[data-test="companyNameTitle"').should("contain", storedCompany.companyInformation.companyName);
+}
```

Comment:
This element is already there on the view-page itself.
I think it would be better to check for the URL instead, and assure that you have been forwarded to the actual edit page

So URL should match with:
`dataland.com/companies/<companyId>/frameworks/lksg/upload?*`

Reply:
The idea was to also test the click functionality form the dropdown. Once we do the click() it automatically goes to the details page... so naturally I check that the company name is the same which also tells us the click worked.

Reply:
Yes your intention makes totally sense. 

But checking the company title is not a safe indicator that the automatic redirect worked in this specific case here.
Reason:
The company title is already shown on the view-page, before you click on "EDIT". So if the "EDIT" button is somehow broken again in the future and does not redirect you after you click on it
`cy.get('[data-test="companyNameTitle"').should("contain", storedCompany.companyInformation.companyName)`
will still pass and the test will not detect the bug.

That's why I think checking for the URL of the expected final destination after the click on "EDIT" might be a better indicator that the redirect works as expected.

Reply:
👍